kaggle_environments/agent.py

```diff
@@ -163,24 +163,32 @@ def act(self, observation):
 
         # Start the timer.
 
-        with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
-            try:
-                start = perf_counter()
-                action = self.agent(*args)
-            except Exception as e:
-                traceback.print_exc(file=err_buffer)
-                action = e
-
-            out = out_buffer.getvalue()
-            err = err_buffer.getvalue()
-            # Get the maximum log length
-            # Allow up to 1k (default) log characters per step which is ~1MB per 600 step episode
-            max_log_length = self.configuration.get('maxLogLength', 1024)
-
-            # truncate if max_log_length is set to None, do not truncate
-            if max_log_length is not None:
-                out = out[0:max_log_length]
-                err = err[0:max_log_length]
+        if self.debug:
+            # Adding a debugging branch here, since the context manager and try except would prevent
+            # debugger from functioning properly.
+            start = perf_counter()
+            action = self.agent(*args)
+            out = ""
+            err = ""
+        else:
+            with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
+                try:
+                    start = perf_counter()
+                    action = self.agent(*args)
+                except Exception as e:
+                    traceback.print_exc(file=err_buffer)
+                    action = e
+
+                out = out_buffer.getvalue()
+                err = err_buffer.getvalue()
+                # Get the maximum log length
+                # Allow up to 1k (default) log characters per step which is ~1MB per 600 step episode
+                max_log_length = self.configuration.get('maxLogLength', 1024)
```

Comment:
note this is updated to 10_000 now by default

Reply:
Sounds good. Will rebase before I merge.